app/ground/receiver_module/src/c_lora_transmit_tenant.cpp

```diff
@@ -0,0 +1,51 @@
+#include "c_lora_transmit_tenant.h"
+#include "c_receiver_module.h"
+
+#include <array>
+#include <n_autocoder_network_defs.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(CLoraTransmitTenant);
+
+void CLoraTransmitTenant::Startup() {
+}
+
+void CLoraTransmitTenant::PostStartup() {
+    // Nothing to do here
+}
+
+void CLoraTransmitTenant::Run() {
+    NTypes::RadioBroadcastData data{};
+    if (readTransmitQueue(data)) {
+        transmit(data);
+    }
+}
+void CLoraTransmitTenant::transmit(const NTypes::RadioBroadcastData& data) const {
+    std::array<uint8_t, 256> txData{};
+
+    if (data.size > (256 - 2)) {
+        // This case should never occur. If it does, then developer is sending too much data
+        LOG_ERR("Received data exceeds LoRa packet size from port %d", data.port);
+        k_oops();
```

Comment:
Is this worth k oopsing for? like kinda in the sense that you wan't them to notice that their data is being truncated but if a situation occurs when the developer doesnt realize until day of that theyre sending too much data in some situations that would be a real awkward way for it to all fail

Reply:
Very fair